extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/schema/001-create-schema.sql

```diff
@@ -0,0 +1,558 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * List for permission data type.
+ */
+CREATE RULE [dbo].[guacamole_permission_list] 
+    AS
+    @list IN ('READ','UPDATE','DELETE','ADMINISTER')
+GO
+
+/**
+ * List for system permission data type.
+ */
+CREATE RULE [dbo].[guacamole_system_permission_list] 
+    AS
+    @list IN ('CREATE_CONNECTION',
+        'CREATE_CONNECTION_GROUP',
+        'CREATE_SHARING_PROFILE',
+        'CREATE_USER',
+        'ADMINISTER')
+GO
+
+/**
+ * The permission data type.
+ */
+CREATE TYPE [dbo].[guacamole_permission] FROM [nvarchar](10) NOT NULL
+
+/**
+ * The system permission data type.
+ */
+CREATE TYPE [dbo].[guacamole_system_permission] FROM [nvarchar](32) NOT NULL
+GO
+
+/**
+ * The connection_group table stores organizational and balancing groups.
+ */
+SET ANSI_NULLS ON
+SET QUOTED_IDENTIFIER ON
+CREATE TABLE [dbo].[guacamole_connection_group](
+    [connection_group_id] [int] IDENTITY(1,1) NOT NULL,
+    [parent_id] [int] NULL,
+    [connection_group_name] [nvarchar](128) NOT NULL,
+    [type] [nvarchar](32) NOT NULL,
+    [max_connections] [int] NULL,
+    [max_connections_per_user] [int] NULL,
+    [enable_session_affinity] [bit] NOT NULL,
+
+    CONSTRAINT [PK_guacmaole_connection_group] PRIMARY KEY CLUSTERED
+        ([connection_group_id] ASC)
+        WITH (PAD_INDEX = OFF,
+            STATISTICS_NORECOMPUTE = OFF,
+            IGNORE_DUP_KEY = OFF,
+            ALLOW_ROW_LOCKS = ON,
+            ALLOW_PAGE_LOCKS = ON)
+       ON [PRIMARY]
+) ON [PRIMARY]

```

Comment:
Though older SQL Server may not require semicolons in all cases, newer versions do, and it's a bad practice that can lead to difficult-to-track-down bugs.

http://www.dbdelta.com/always-use-semicolon-statement-terminators/

Better that we stick with ANSI as far as possible.

Reply:
Okay, I've added those to the code.  You know, it's almost hard to believe that I was just using Microsoft-generated code - that Microsoft would do something that didn't conform to standards!